spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessorTests.java

```diff
@@ -231,6 +233,22 @@ public void multipleRoutingKeysTestBean() {
 		context.close();
 	}
 
+	@Test
+	public void customExhangeTestBean() {
+		ConfigurableApplicationContext context = new AnnotationConfigApplicationContext(Config.class,
+				CustomExchangeTestBean.class);
+
+		final Collection<org.springframework.amqp.core.Exchange> exchanges = context
+				.getBeansOfType(org.springframework.amqp.core.Exchange.class).values();
+		assertThat(exchanges, hasSize(1));
+		assertThat(exchanges.iterator().next(), instanceOf(CustomExchange.class));
+		final CustomExchange customExchange = (CustomExchange) exchanges.iterator().next();
+		assertEquals("my_custom_exchange", customExchange.getName());
+		assertEquals("custom_type", customExchange.getType());
+
+		context.close();
```

Comment:
From big height the fix looks reasonable for me.
But does it work really against real Broker?
I'm just curious if that `CustomExchange` is robust.

But, yeah, thank you @panchenko for the effort to support community and improve the Framework!